### PR TITLE
Starter Page Templates: Add all SPT themes home templates to the template selector

### LINKED ISSUE #37282: Editor: Add homepage templates from all template first themes to page template selector
We have many homepage templates built for our 18 template-first Gutenberg themes. All of these page templates should be available in the template picker, regardless of the theme that you have chosen.

Currently, users think they have to change their entire theme in order to get a different home page template. They're correct, but this should not be necessary.

Ref: p58i-8cp-wpcom

### PR DESCRIPTION
**Pull request by Copons**
#### Changes proposed in this Pull Request

* Update SPT to request all SPT themes homepage templates.
* Display these homepage templates in a new additional grouping in the SPT selector.

| Before | After |
| - | - |
| Screenshot 2019-11-07 at 16 46 33 | Screenshot 2019-11-08 at 12 25 14 |

#### Testing instructions

* Apply D35170-code and sandbox the API.
* Build this PR with `npx lerna run build --scope='@automattic/full-site-editing'` on a site with SPT enabled (a local FSE test environment is fine, but it might lack some blocks required by the theme).
* Create a new page on that site and make sure it shows the Starter Page Templates selector.
* Make sure it has both the expected 9 vertical templates (Blank, Home, etc.), and a second grouping with many other templates named after their themes (e.g. Mayland, Rivington, Dalston, etc.).

Fixes #37282

### CODE REVIEW

**Comment by shaunandrews** [comment]
I think the "Or choose a homepage..." label is way too wordy, and the spacing above it is too small.

Can we match the wording and spacing (42px) from the design:

**Comment by apeatling** [comment]
+1 to @shaunandrews revisions. Let's definitely avoid the word "theme" here.

**Comment by apeatling** [comment]
@shaunandrews Once we get this change deployed, I think we need more thinking on what the next revision looks like. Grouping templates by type/goal for example. Think Apple stacks.

**Comment by Copons** [comment]
@shaunandrews @apeatling I'm so sorry, I forgot to mention that this PR purpose is mostly to see if the backend changes make sense.

> Also, I noticed that the "Home" template still lives in the top group — it should be in the group with the other homepage templates.

Ah, I totally didn't notice this!
I just assumed we wanted the current behaviour + a second group with all _other_ home templates.
I'll update the endpoint for this.

**Comment by Copons** [comment]
Updated the PR for the latest backend changes, and with all the suggestions posted here.

Screenshot 2019-11-08 at 12 25 14

File: apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php [hunk 1]
```diff
@@ -205,10 +205,7 @@ public function fetch_vertical_data() {
 		if ( false === $vertical_templates || ( defined( 'WP_DEBUG' ) && WP_DEBUG ) ) {
 			$vertical_id = get_option( 'site_vertical', 'default' );
 			$request_url = add_query_arg(
-				[
-					'_locale' => $this->get_iso_639_locale(),
-					'theme'   => get_stylesheet(),
-				],
+				[ '_locale' => $this->get_iso_639_locale() ],
```
Comment (obenland) [inline apps/full-site-editing/full-site-editing-plugin/starter-page-templates/class-starter-page-templates.php]: Nice

**Comment by Copons** [comment]
Note: I've put this on hold pending backend work.

**Comment by apeatling** [comment]
@Copons What backend work is needed?

**Comment by Copons** [comment]
@apeatling the diff works as it is, so worst case scenario we can just ship, but the performances are not there yet.
We have to figure out if there's a way speed up the themes list (`wp_get_themes`) somehow, maybe caching it. Hopefully someone else already wrote a solution for this though!

**Comment by Copons** [comment]
> (non-blocking) Do you think it would make sense to to make the preview position fixed for larger screens too, so that we don't have to scroll back up after selecting the template at the bottom? It seems like this behavior is already in place for widths smaller than 1440px, and I'm not sure why it doesn't apply in general.

@vindl That seems a bit complicated to fix, I guess we could tackle it in a follow-up, or actually just wait for the [new design](https://github.com/Automattic/wp-calypso/issues/37282#issuecomment-553531255) to be finalized. 🙂

**Comment by apeatling** [comment]
Tested this, and I think this is fine to ship when the backend patch is ready, but please follow up on the fixed position main preview window once it's merged. Nice work @Copons!

**Comment by apeatling** [comment]
Many of the homepage templates also look broken in the preview, we need to loop back on that too.

**Comment by shaunandrews** [comment]
In Core, we recently discussed avoiding the term "Template" in favor of "Layout." I don't consider it a blocker for this, but if we could change the term we could see how people perceive it in our next testing session.

**Comment by apeatling** [comment]
@Copons I actually think we should change the wording from template to layout in this since it's already been done here: https://github.com/Automattic/wp-calypso/pull/37500

**Comment by noahtallen** [comment]
Btw @shaunandrews @Copons I have this PR: https://github.com/Automattic/wp-calypso/pull/37624 which changes "Template" to "Layout" for cases outside of this PR.

**Comment by apeatling** [comment]
🎉👏
